library/alloc/src/macros.rs

```diff
@@ -29,6 +29,9 @@
 /// to the same boxed integer value, not five references pointing to independently
 /// boxed integers.
 ///
+/// Also, note that `[T; 0]` is a valid initializer. This will initialize (or call)
+/// `T` but not populate the vector with it, so be mindful of side effects.
```

Comment:
I think you mean `vec![T; 0]` here?

Also I would suggest not using `T` for an expression since it's generally used for a type.

Maybe something like
> Also, note that `vec![expr; 0]` is allowed, and produces an empty vector.  This will still evaluate `expr`, however, and immediately drop the resulting value, so be mindful of side effects.

(And similarly below.)

Reply:
This is in the `array` docs, so I intentionally dropped the `vec!` bit. 

> Also I would suggest not using `T` for an expression since it's generally used for a type.
> 
> Maybe something like
> 
> > Also, note that `vec![expr; 0]` is allowed, and produces an empty vector.  This will still evaluate `expr`, however, and immediately drop the resulting value, so be mindful of side effects.
> 
> (And similarly below.)

Sounds good to me! I'll make those changes.

Reply:
> This is in the array docs, so I intentionally dropped the vec! bit.

In the other file yes, but this file is in `alloc` and is the docs for `macro_rules! vec {`.

Reply:
Sorry, you're right. I'll also fix that.

Reply:
Fixes applied. Let me know what you think!